card/src/main/java/com/adyen/checkout/card/CardValidationUtils.kt

```diff
@@ -94,17 +96,17 @@ object CardValidationUtils {
 
     /**
      * Validate Security Code.
-     * We always pass CardType null, but we can enforce size validation for Amex or otherwise if necessary.
      */
-    fun validateSecurityCode(securityCode: String, cardType: CardType?): FieldState<String> {
+    fun validateSecurityCode(securityCode: String, cardType: DetectedCardType?): FieldState<String> {
         val normalizedSecurityCode = StringUtil.normalize(securityCode)
         val length = normalizedSecurityCode.length
         var validation: Validation = Validation.Invalid(R.string.checkout_security_code_not_valid)
         if (StringUtil.isDigitsAndSeparatorsOnly(normalizedSecurityCode)) {
-            if (cardType == CardType.AMERICAN_EXPRESS && length == AMEX_SECURITY_CODE_SIZE) {
-                validation = Validation.Valid
-            } else if (length == GENERAL_CARD_SECURITY_CODE_SIZE && cardType != CardType.AMERICAN_EXPRESS) {
-                validation = Validation.Valid
+            validation = when {
+                cardType?.cvcPolicy == Brand.CvcPolicy.OPTIONAL && length == 0 -> Validation.Valid
```

Comment:
Added `length == 0` check here to make sure we don't consider ANY optional CVC as valid, but only if it's empty. Otherwise it must pass through the normal validation steps.